toolchain/sem_ir/inst_namer.cpp

```diff
@@ -767,15 +767,15 @@ auto InstNamer::NamingContext::AddIntOrFloatTypeName(char type_literal_prefix,
 auto InstNamer::NamingContext::AddWitnessTableName(InstId witness_table_inst_id,
                                                    std::string name) -> void {
   auto witness_table =
-      sem_ir().insts().GetAs<ImplWitnessTable>(witness_table_inst_id);
-  if (!witness_table.impl_id.has_value()) {
+      sem_ir().insts().TryGetAs<ImplWitnessTable>(witness_table_inst_id);
+  if (!witness_table || !witness_table->impl_id.has_value()) {
     // TODO: The witness comes from a facet value. Can we get the
     // interface names from it? Store the facet value instruction in the
     // table?
     AddInstName(name);
     return;
   }
-  const auto& impl = sem_ir().impls().Get(witness_table.impl_id);
+  const auto& impl = sem_ir().impls().Get(witness_table->impl_id);
   auto name_id = sem_ir().interfaces().Get(impl.interface.interface_id).name_id;
```

Comment:
This function is called for an `ImplWitness` instruction, which we're not using now. Are these changes still neeed?

Reply:
No, you're right, these ended up being unrelated to this PR. Generally I've been on the side of making the formatter and inst namer robust in the presence of invalid IR, though, since they're used in part as debugging tools. Given that context, I can leave this as-is, split this out to a separate PR or just discard it, what would you prefer?

Reply:
One of the callers is doing that check; the other one is just assuming that the table in an `ImplWitness` will always be an `ImplWitnessTable` instruction (which crashed here when I changed it to sometimes be a `CppWitnessTable` in an earlier revision of this patch). I'll split this out into a separate PR with the other cleanup and we can discuss there whether we want this or not.